osidb/models/affect.py

```diff
@@ -190,14 +190,33 @@ class Meta:
     def __str__(self):
         return str(self.uuid)
 
+    def ps_component_from_purl(self, should_raise=False):
+        try:
+            # try to parse the PS component from the PURL but do not raise any
+            # error on failure as that will be done as part of the validations
+            purl = PackageURL.from_string(self.purl)
+            if purl.type == "oci":
+                try:
+                    prefix = purl.qualifiers["repository_url"].split("/")[1]
+                    return f"{prefix}/{purl.name}"
```

Comment:
I'm not sure about the Error handling code here. But the component_name return for an oci purl looks good to me.

Reply:
I'm not sure either, but its awkwardness might be a feature (it wants to be rewritten or deleted!) if we wanted to remove it later. It's also to work with the validator logic that comes later in the class.